### PR TITLE
Improve SeekBarMapper

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Improve SeekBarMapper and its unit tests: 

- Better support for padding
- Better support for thumb size
- Include background if any
- More readable tests

### CODE REVIEW

File: features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseSeekBarWireframeMapperTest.kt [hunk 1]
```diff
@@ -151,6 +142,7 @@ internal abstract class BaseSeekBarWireframeMapperTest {
     @Mock
     lateinit var mockInternalLogger: InternalLogger
 
+    /*
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseSeekBarWireframeMapperTest.kt:145]: should it be left as commented out?

File: features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/mapper/AbstractWireframeMapperTest.kt [hunk 2]
```diff
@@ -82,22 +101,75 @@ internal abstract class AbstractWireframeMapperTest<V : View, WM : WireframeMapp
...
+        assertThat(actual.y).isCloseTo(expected.y, withinPercentage(5))
+        assertThat(actual.width).isCloseTo(expected.width, withinPercentage(5))
+        assertThat(actual.height).isCloseTo(expected.height, withinPercentage(5))
+
+        val actualCornerRadius = actual.shapeStyle?.cornerRadius?.toLong()
+        val expectedCornerRadius = expected.shapeStyle?.cornerRadius?.toLong()
+        if (actualCornerRadius != null && expectedCornerRadius != null) {
+            assertThat(actualCornerRadius).isCloseTo(expectedCornerRadius, withinPercentage(5))
+        }
+    }
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/mapper/AbstractWireframeMapperTest.kt:174]: `// endregion` is missing, it seems

File: features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/LegacySeekBarWireframeMapperTest.kt [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(ForgeConfigurator::class)
+internal class LegacySeekBarWireframeMapperTest : BaseSeekBarWireframeMapperTest() {
+
+    /*
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/LegacySeekBarWireframeMapperTest.kt:26]: should it all be commented out?

File: features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseSeekBarWireframeMapperTest.kt (hunk 1 above)
Comment (xgouchet) [reply features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseSeekBarWireframeMapperTest.kt:145]: Good call, this code should be deleted

File: features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/LegacySeekBarWireframeMapperTest.kt (hunk 3 above)
Comment (xgouchet) [reply features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/LegacySeekBarWireframeMapperTest.kt:26]: Same, it's deleted

File: features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/mapper/AbstractWireframeMapperTest.kt [hunk 4]
```diff
@@ -82,22 +101,75 @@ internal abstract class AbstractWireframeMapperTest<V : View, WM : WireframeMapp
...
+        whenever(mockViewBoundsResolver.resolveViewPaddedBounds(mock, fakeDensity)) doReturn fakeViewPaddedBounds
         whenever(mockViewIdentifierResolver.resolveViewId(mock)) doReturn fakeViewIdentifier
 
         mockMappedView = mock
     }
 
     // endregion
+
+    // region Assertions
+    fun assertThatBoundsAreCloseEnough(actual: MobileSegment.Wireframe, expected: MobileSegment.Wireframe) {
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/mapper/AbstractWireframeMapperTest.kt:147]: 👍

File: features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/mapper/AbstractWireframeMapperTest.kt (hunk 2 above)
Comment (xgouchet) [reply features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/mapper/AbstractWireframeMapperTest.kt:174]: 👍

**Review by 0xnm — APPROVED** [review]

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/utils/ColorConstants.kt [hunk 5]
```diff
@@ -8,10 +8,16 @@ package com.datadog.android.sessionreplay.utils
...
 
 internal const val MASK_ALPHA = 0xFF000000L
 internal const val MASK_RGB = 0x00FFFFFFL
 internal const val MASK_COLOR = 0xFFFFFFFFL
+
+/** The value corresponding to an opaque Alpha. */
+const val OPAQUE_ALPHA_VALUE: Int = MAX_ALPHA_VALUE
+
+/** The value corresponding to a 25% opaque Alpha. */
+const val PARTIALLY_OPAQUE_ALPHA_VALUE: Int = 0x44
```
Comment (ambushwork) [inline features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/utils/ColorConstants.kt:23]: a random mathematical comment here, 0x3f or 0x40 seems closer to 25% of opaque alpha.(Just wondering how 0x44 is calculated, although it hardly has any impact)

Comment (xgouchet) [reply features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/utils/ColorConstants.kt]: Good point thanks

**Pull request merged**